help/share-and-upload-files.md

```diff
@@ -1,21 +1,103 @@
 # Share and upload files
 
-Attach files to messages, including images, documents, sound, and video.
+Zulip supports attaching multiple files to messages, including images,
+documents, sound, and video. You can edit the names of the files others see
+after you upload them.
 
-First, [open the compose box](/help/open-the-compose-box). Then
+Zulip will automatically generate a **thumbnail** for each file when you send
+the message, if it can. Image thumbnails will be shown directly in the message,
+and you can click on a thumbnail to [view the full image](/help/view-and-browse-images).
 
-* **Drag and drop** files into the compose box.
-* **Copy and paste** files into the compose box.
-* Click the **paperclip** (<i class="fa fa-paperclip"></i>) icon at
-  the bottom of the compose box to find files on your computer.
+## Uploading files
 
-Zulip will insert a link to the file, in Markdown format:
-`[link text](URL)`. You can modify the `link text` to whatever you want.
+{start_tabs}
 
-Zulip will automatically generate a **thumbnail** of the file when you send
-it, if it can. You can
-[preview the message](/help/preview-your-message-before-sending) before
-sending to see what the thumbnail will look like.
+{tab|via-markdown}
+
+{!start-composing.md!}
+
+1. Drag and drop files, or copy and paste one or more files into the compose
+   box. Zulip will upload the files, and insert named links using
+   [Markdown formatting](/help/format-your-message-using-markdown#links):
+   `[Link text](URL)`.
+
+1. _(optional)_ Modify the link text as desired.
+
+!!! tip ""
+
+    You can [preview the message](/help/preview-your-message-before-sending)
+    before sending to see what your uploaded files will look like.
+
+{tab|via-compose-box-buttons}
+
+{!start-composing.md!}
+
+1. Click the **paperclip** (<i class="fa fa-paperclip"></i>) icon at
+   the bottom of the compose box to select one or more files. Zulip will upload
+   the files, and insert named links using
+   [Markdown formatting](/help/format-your-message-using-markdown#links):
+   `[Link text](URL)`.
+
+1. _(optional)_ Modify the link text as desired.
+
+!!! tip ""
+
+    You can [preview the message](/help/preview-your-message-before-sending)
+    before sending to see what your uploaded files will look like.
+
+{tab|mobile}
+
+1. Navigate to a stream, topic, or direct message view.
+
+1. Tap the
+   **paperclip** (<img src="/static/images/help/mobile-paperclip-icon.svg" alt="paperclip" class="mobile-icon"/>),
+   **image** (<img src="/static/images/help/mobile-image-icon.svg" alt="image" class="mobile-icon"/>),
+   or **camera** (<img src="/static/images/help/mobile-camera-icon.svg" alt="camera" class="mobile-icon"/>)
+   button at the bottom of the app to select one or more files. Zulip will
```

Comment:
This is a bit confusing because it leaves it a mystery why there are these three different buttons, instead of just one.

Here's a description of the three (but it can probably be tightened for this context):
* Paperclip: Choose an arbitrary file on your device, using your device's file browser.
* Image: Choose an image from your device's image gallery.
* Camera: Take a photo with your device's camera, and share that.

In particular if you wanted to "select a file" and you use the camera button, you are definitely going to be disappointed because it won't offer you any existing files — its job is to help you take a new photo to share.